apps/drain-safe/src/components/App.tsx

```diff
@@ -108,13 +114,52 @@ const App: React.FC = () => {
     }
   }, [balancesError]);
 
+  useEffect(() => {
```

Comment:
What do you think about creating a useWeb3 hook? for avoid passing down props to de final CurrencyCell. You can gather the chains.ts info there as well as we are going to remove it later

Reply:
I think that it's a good idea to create a `useWeb3` hook, but I think that we should keep it in the `App.tsx` because we only want instantiate web3 and call `getGasPrice` only once and not in each `CurrencyCell`.

Reply:
+1 for a separate hook, I find that it vastly improves readability

Reply:
Could perhaps be a separate ticket @dasanra ? as TX Builder and Drain Safe can share the same hook and as @DaniSomoza mentioned me offline, we will need a context provider as well for avoid instance recreation.

Reply:
Why would we want to avoid it? In the drain-safe app, it's only used in one component and then passed down via props. To me, it looks like just moving it to a separate file

Reply:
For avoid the prop to only exist to be sent down. In this case in the Balances I think

Reply:
but the context is fine isn't it ? I mean, a context for the web3 instance and a hook to instantiate it using our getChainInfo populated with RPC url (task in progress) could be helpful for retrieving it in any level of the component tree

Reply:
done! Added useWeb3.js hook